src/resolve/resolved-dns-packet.h

```diff
@@ -81,6 +81,9 @@ struct DnsPacket {
         /* For support of truncated packets */
         DnsPacket *more;
 
+        /* For support of packets in queue */
+        LIST_FIELDS(DnsPacket, packets_in_queue);
+
```

Comment:
hmm, I wonder if this is a safe thing to do. So far we considered packets more or less "immutable" after they have been created once, thus encouraging that consumers just add refs if they want to keep them around and don't have to be afraid that they change under their accesses too wildly. Coding using them can do with the packets whatever it wants really, in the current code and in future code and as long as they take a ref they know that the packet is gonna stay pinned and remain ina good state.

However, this property is somewhat affected by this embedded chain, as this means the packets can be used in a specific context only once: you can enqueue it in exactly one DnsStream queue exactly once. If any other code gets ahold of the DnsPacket, then it can't use it in that context really, and if it does by some accident weird things will happen…

Now, in the current code I can't see why one ever would want to enqueue the same DnsPacket in more than one stream, or more than once in the same stream, but I still would prefer if we'd not embedd the linked list structure here, just to keep the code more versatile and generic for future changes.

Hence, instead of embedding a linked list here, I think it would be nice to just use an `OrderedSet` object for maintaining the queue of packets to write. I figure the code is probably going to be shorter too that way, and a lot more generic, as enqueing the same DnsPacket objects into multiple streams or into the same stream more than once would be fully safe then.